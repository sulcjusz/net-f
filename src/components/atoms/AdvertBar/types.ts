export interface IAdvertBarProps {
  children: React.ReactNode
  className?: string
}
