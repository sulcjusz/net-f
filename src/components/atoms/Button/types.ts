export interface IButtonProps {
  onClick?: () => void
  size?: 'small' | 'medium'
  disabled?: boolean
  children: React.ReactNode
  className?: string
}
