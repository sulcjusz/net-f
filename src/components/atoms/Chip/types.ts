export interface IChipProps {
  size?: 'small' | 'medium'
  textColor?: 'black' | 'white'
  children: React.ReactNode
  className?: string
}
