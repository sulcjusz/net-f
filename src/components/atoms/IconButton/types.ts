export interface IIconButtonProps {
  onClick?: () => void
  size?: 'small' | 'medium'
  disabled?: boolean
  icon: React.ReactNode
  className?: string
}
