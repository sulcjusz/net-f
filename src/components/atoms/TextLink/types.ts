export interface ITextLinkProps {
  text: string
  onClick: () => void
  className?: string
}
