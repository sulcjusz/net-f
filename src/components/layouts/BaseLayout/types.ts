export interface IBaseLayoutProps {
  children: React.ReactNode
}
