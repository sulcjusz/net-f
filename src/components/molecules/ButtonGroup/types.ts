export interface IButtonGroupProps {
  children: React.ReactNode
}
