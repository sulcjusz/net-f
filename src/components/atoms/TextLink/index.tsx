import React from 'react'
import classnames from 'classnames'
import {ITextLinkProps} from './types'

const TextLink: React.FC<ITextLinkProps> = ({text, onClick, className}) => {
  const linkClasses = classnames('text-blue-800 cursor-pointer', className)

  return (
    <span className={linkClasses} onClick={onClick}>
      {text}
    </span>
  )
}

export default TextLink
