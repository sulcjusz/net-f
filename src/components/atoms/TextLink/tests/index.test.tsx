import {render, fireEvent, RenderResult} from '@testing-library/react'
import '@testing-library/jest-dom'
import TextLink from '../index'
import {ITextLinkProps} from '../types'

type RenderTextLinkProps = Partial<ITextLinkProps>

const renderTextLink = (props: RenderTextLinkProps = {}): RenderResult => {
  const {text, onClick, className} = props
  return render(
    <TextLink
      text={text || 'Link'}
      onClick={onClick || jest.fn()}
      className={className}
    />
  )
}

describe('TextLink component', () => {
  it('matches snapshot', () => {
    const {container} = renderTextLink({text: 'Snapshot Test'})
    expect(container).toMatchSnapshot()
  })

  it('renders text correctly', () => {
    const text = 'Click me'
    const {getByText} = renderTextLink({text})
    expect(getByText(text)).toBeInTheDocument()
  })

  it('renders with default class', () => {
    const {getByText} = renderTextLink()
    expect(getByText('Link')).toHaveClass('text-blue-dark')
    expect(getByText('Link')).toHaveClass('cursor-pointer')
  })

  it('renders with additional className', () => {
    const {getByText} = renderTextLink({className: 'custom-class'})
    expect(getByText('Link')).toHaveClass('custom-class')
  })

  it('calls onClick handler when clicked', () => {
    const onClickMock = jest.fn()
    const {getByText} = renderTextLink({onClick: onClickMock})
    fireEvent.click(getByText('Link'))
    expect(onClickMock).toHaveBeenCalledTimes(1)
  })
})
