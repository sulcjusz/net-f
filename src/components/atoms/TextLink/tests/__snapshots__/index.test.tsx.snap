// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLink component matches snapshot 1`] = `
<div>
  <span
    class="text-blue-dark cursor-pointer"
  >
    Snapshot Test
  </span>
</div>
`;
